miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebViewClient.kt

```diff
@@ -28,17 +28,16 @@ internal class MiniAppWebViewClient(
         return response
     }
 
-    override fun shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest): Boolean {
-        var shouldCancelLoading = super.shouldOverrideUrlLoading(view, request)
-        if (request.url != null) {
-            val requestUrl = request.url.toString()
-            if (requestUrl.startsWith("tel:")) {
-                openPhoneDialer(requestUrl)
+    override fun shouldOverrideUrlLoading(view: WebView?, url: String?): Boolean {
```

Comment:
According to the [official doc of WebViewClient](https://developer.android.com/reference/android/webkit/WebViewClient), this method has been deprecated in API level `24`.  But, this is the working function below of `24`, I think, it's a bug itself inside Android SDK.. 🤔 

My idea is using latest `shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest)` for API level `24+`, and using `shouldOverrideUrlLoading(view: WebView?, url: String?)` for below than `24`. 

So, we can refactor this part as following:

```
    @SuppressWarnings("deprecation")
    override fun shouldOverrideUrlLoading(view: WebView?, url: String?): Boolean {
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N)
            return overrideUrlLoading(url.toString(), super.shouldOverrideUrlLoading(view, url))

        return false
    }

    @RequiresApi(Build.VERSION_CODES.N)
    override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest): Boolean {
        return overrideUrlLoading(request.url.toString(), super.shouldOverrideUrlLoading(view, request))
    }

    private fun overrideUrlLoading(requestUrl: String, shouldOverride: Boolean): Boolean {
        var shouldCancelLoading = shouldOverride
        if (requestUrl.startsWith("tel:")) {
            openPhoneDialer(requestUrl)
            shouldCancelLoading = true
        } else if (....) {
           ...
        }
        return shouldCancelLoading
    }
```

Reply:
It's a good idea but I think we can omit this version checking by using only one `shouldOverrideUrlLoading`. I tested and it's working fine on other android versions.
We may update our sdk to 24 as minimum later so let's keep this fix simple.

Reply:
Yes, we need to use `override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest)` when we can update the sdk to 24 as minimum level. For now, the current change is also accepted, so it's approved from my side 👍

Reply:
@minh-rakuten Can you add a `TODO` here so we remember to update this in the future?

Reply:
ToDo has been added.